src/R/Components/Impl/PackageManager/Implementation/ViewModel/RPackageManagerViewModel.cs

```diff
@@ -127,77 +134,163 @@ internal class RPackageManagerViewModel : BindableBase, IRPackageManagerViewMode
 
         private async Task SwitchToAvailablePackagesAsync() {
             _coreShell.AssertIsOnMainThread();
-            await EnsureInstalledAndAvailablePackagesLoadedAsync();
+            if (_selectedTab == SelectedTab.AvailablePackages) {
+                IsLoading = !_availableLock.IsCompleted;
+            }
+            
+            await EnsureAvailablePackagesLoadedAsync();
+            if (_selectedTab == SelectedTab.AvailablePackages) {
+                IsLoading = false;
+            }
+
             if (_availablePackages == null) {
                 return;
             }
             _items.ReplaceWith(_availablePackages);
         }
 
+        private async Task EnsureAvailablePackagesLoadedAsync() {
```

Comment:
It looks like you could switch to bg thread here already - nothing in this needs UI.